rmw_opendds_cpp/src/OpenDDSNode.cpp

```diff
@@ -119,15 +127,221 @@ rmw_ret_t OpenDDSNode::count_subscribers(const char * topic_name, size_t * count
   return RMW_RET_OK;
 }
 
-OpenDDSNode::OpenDDSNode(rmw_context_t & context)
+rmw_ret_t OpenDDSNode::get_names(rcutils_string_array_t * names, rcutils_string_array_t * namespaces, rcutils_string_array_t * enclaves) const
+{
+  if (rmw_check_zero_rmw_string_array(names) != RMW_RET_OK) {
+    RMW_SET_ERROR_MSG("rmw_check_zero_rmw_string_array(names) failed.");
+    return RMW_RET_INVALID_ARGUMENT;
+  }
+  if (rmw_check_zero_rmw_string_array(namespaces) != RMW_RET_OK) {
+    RMW_SET_ERROR_MSG("rmw_check_zero_rmw_string_array(namespaces) failed.");
+    return RMW_RET_INVALID_ARGUMENT;
+  }
+  DDS::InstanceHandleSeq handles;
+  if (dp_->get_discovered_participants(handles) != DDS::RETCODE_OK) {
+    RMW_SET_ERROR_MSG("unable to fetch discovered participants.");
+    return RMW_RET_ERROR;
+  }
+  const auto length = handles.length() + 1;  // add yourself
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  try {
+    if (rcutils_string_array_init(names, length, &allocator) != RCUTILS_RET_OK) {
+      throw std::runtime_error(rcutils_get_error_string().str);
+    }
+    if (rcutils_string_array_init(namespaces, length, &allocator) != RCUTILS_RET_OK) {
+      throw std::runtime_error(rcutils_get_error_string().str);
+    }
+    if (enclaves) {
+      if (rcutils_string_array_init(enclaves, length, &allocator) != RCUTILS_RET_OK) {
+        throw std::runtime_error(rcutils_get_error_string().str);
+      }
+    }
+  //?? why data[0] cannot be put in the for loop?
+  //?? names->data[0] not set?
```

Comment:
Are we waiting for answers here?

Reply:
These were the things I found when moving and simplifying the existing code. They should be in a separate PR to determine whether the processing logic here is correct or needs to be modified. It's a safer strategy to have the main framework changes verified first, then to investigate such unobvious issues.

Reply:
I think it's supposed to be like this (https://github.com/ros2/rmw_connext/blob/1.0.1/rmw_connext_shared_cpp/src/node_names.cpp#L111-L131) since this is the code it was created from.

Reply:
added node names->data[0]